### DISCUSSION TITLE
Modification of attachment body in plugin

### DISCUSSION THREAD

**Discussion by biski** [allure-framework/allure2#2275] [Questions & Support] [answered]
Hi, 
we have a plugin which modifies contents of attachments. We do this because we need to have in some attachment a link to the current test case and link to the screenshot from the current test case.
I wrote a plugin which search attachments and replaces tag #TESTCASE# with current test case id (TestResult.getUid()). It was working well in version 2.15.0. Unfortunately, after upgrade to 2.25.0 it doesn't work. I was using Aggregator to achieve this behavior. I noticed that outputDirectory no longer points to allure-reports folder but to /tmp/allure-reports which is empty. Furthermore, I also tried to use Aggregator2, but using this class I don't have access to folder directory at all.

Is it a bug? Could you give me a workaround? 
Thanks

**Comment by baev** [comment, marked as answer]
We removed the possibility to access the output directory when we [introduced single-file mode](https://github.com/allure-framework/allure2/pull/2072). 

If you want to override the continents of attachments, you can override the data file by calling `storage.addDataFile` with the same data key. 

```java
@Override
public void aggregate(final Configuration configuration,
                      final List<LaunchResults> launchesResults,
                      final ReportStorage storage) {
    launchesResults.forEach(launch -> launch.getAttachments().entrySet()
            .parallelStream()
            .forEach(entry -> {
                final String dataKey = Constants.dataPath("attachments", entry.getValue().getSource());
                final Path content = entry.getKey();
                final Path modifiedContent = modify(content);
                storage.addDataFile(
                    dataKey,
                    modifiedContent
                );
            }));
}
```

**Reply by biski** [reply]
Thanks
